tests/integration/helm/upgrade/util.go

```diff
@@ -139,8 +140,25 @@ func performInPlaceUpgradeFunc(previousVersion string, isAmbient bool) func(fram
 			// all versions
 			helmtest.DeleteIstio(t, h, cs, nsConfig, isAmbient)
 		})
+		t.Cleanup(func() {
```

Comment:
Hmm, when I looked briefly at this, it looked like the underlying test framework already unconditionally called the `cleanup` func which dumps the istio namespace if `t.Failed() && t.Setttings().CIMode` by default - even for Helm tests

If you've tested this works locally tho, nvm.

Reply:
https://github.com/istio/istio/issues/55215#issuecomment-2703742743

Ah nvm you addressed this already, OK.